gelf/assembler.go

```diff
@@ -0,0 +1,56 @@
+package gelf
+
+import (
+	"bytes"
+	"time"
+)
+
+// Assembler provides GELF message de-chunking
+type assembler struct {
+	deadline   time.Time
+	chunksData [][]byte
+	processed  int
+	totalBytes int
+}
+
+// NewAssembler returns empty Assembler with maximum message size and duration
+// specified.
+func newAssembler(timeout time.Duration) (res *assembler) {
+	res = new(assembler)
+	res.deadline = time.Now().Add(timeout)
+	return
+}
+
+// Bytes returns message bytes, not nessesarily fully assembled
+func (a *assembler) Bytes() []byte {
+	return bytes.Join(a.chunksData, nil)
+}
+
+// Expired returns true if first chunk is too old
+func (a *assembler) Expired() bool {
```

Comment:
Nit: would not export this

```suggestion
func (a *assembler) expired() bool {
```

Reply:
done